app/src/main/java/com/github/houseorganizer/houseorganizer/house/QRCodeScanActivity.java

```diff
@@ -0,0 +1,149 @@
+package com.github.houseorganizer.houseorganizer.house;
+
+import static com.github.houseorganizer.houseorganizer.panels.MainScreenActivity.CURRENT_HOUSEHOLD;
+import static com.github.houseorganizer.houseorganizer.util.Util.getSharedPrefsEditor;
+
+import android.Manifest;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.util.Size;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.camera.core.CameraSelector;
+import androidx.camera.core.ImageAnalysis;
+import androidx.camera.core.Preview;
+import androidx.camera.lifecycle.ProcessCameraProvider;
+import androidx.camera.view.PreviewView;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+
+import com.github.houseorganizer.houseorganizer.R;
+import com.github.houseorganizer.houseorganizer.image.QRAnalyzer;
+import com.github.houseorganizer.houseorganizer.image.QRListener;
+import com.github.houseorganizer.houseorganizer.panels.MainScreenActivity;
+import com.github.houseorganizer.houseorganizer.util.Util;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.FieldValue;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import java.util.List;
+import java.util.Map;
+
+public class QRCodeScanActivity extends AppCompatActivity {
+
+    private FirebaseFirestore db;
+    private FirebaseAuth auth;
+
+    private static final int CAMERA_PERMISSION_REQ_CODE = 1; //Can be chosen but has to be unique
+    private static final String CAMERA_PERMISSION = Manifest.permission.CAMERA;
+
+    private PreviewView cameraPreview;
+    private ListenableFuture<ProcessCameraProvider> cameraProvider;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_qrcode_scan);
+
+        db = FirebaseFirestore.getInstance();
+        auth = FirebaseAuth.getInstance();
+
+        cameraPreview = findViewById(R.id.cameraPreview);
+        cameraProvider = ProcessCameraProvider.getInstance(this);
+
+        askForCameraPermissionsOrContinue();
+    }
+
+    private void askForCameraPermissionsOrContinue() {
```

Comment:
Nice method - will probably be using something similar for the offline mode

Reply:
Yup, seems to work